analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpDiagnosticAnalyzerContextHelper.cs

```diff
@@ -57,17 +57,9 @@ public static void ReportDiagnosticIfNonGenerated(this CompilationAnalysisContex
             }
         }
 
```

Comment:
`RegisterSyntaxTreeActionInNonGenerated` at line 42 is not used anymore and can be removed.

Reply:
It's removed in #6545